libexec/python/shell.py

```diff
@@ -99,63 +187,66 @@ def parse_image_uri(image,
     if default_tag is None:
         default_tag = TAG
 
-    # Be absolutely sure there are not comments
+    # if user gave custom registry / namespace, make them the default
+    if CUSTOM_NAMESPACE is not None:
+        default_namespace = CUSTOM_NAMESPACE
+
+    if CUSTOM_REGISTRY is not None:
+        default_registry = CUSTOM_REGISTRY
+
+    # candidate regex for matching, in order of preference
+    uri_regexes = [ _reduced_uri_no_ns_re,
+                    _default_uri_re ]
+
+    # Be absolutely sure there are no comments
     image = image.split('#')[0]
 
-    # Get rid of any uri, and split the tag
-    image = image.replace(uri, '')
-
-    # Does the uri have a digest or Github tag (version)?
-    image = image.split('@')
-    version = None
-    if len(image) == 2:
-        version = image[1]
-
-    image = image[0]
-    image = image.split(':')
-
-    # If there are three parts, we have port and tag
-    if len(image) == 3:
-        repo_tag = image[2]
-        image = "%s:%s" % (image[0], image[1])
-
-    # If there are two parts, we have port or tag
-    elif len(image) == 2:
-        # If there isn't a slash in second part, we have a tag
-        if image[1].find("/") == -1:
-            repo_tag = image[1]
-            image = image[0]
-        # Otherwise we have a port and we merge the path
-        else:
-            image = "%s:%s" % (image[0], image[1])
-            repo_tag = default_tag
-    else:
-        image = image[0]
-        repo_tag = default_tag
+    if not uri:
+        uri = get_image_uri(image, quiet=True)
 
-    # Now look for registry, namespace, repo
-    image = image.split('/')
+    # docker images require slightly different rules
+    if uri == "docker://":
+        uri_regexes = [ _docker_uri_re ]
 
-    if len(image) > 2:
-        registry = image[0]
-        namespace = "/".join(image[1:-1])
-        repo_name = image[-1]
+    image = remove_image_uri(image, uri);
```

Comment:
I think the ";" snuck in from another language... :O

Reply:
But python accepts it too! :) C'mon I wrote C and C++ code all day for a living, give me a break ... but I'll fix it.

Reply:
haha :P thank you :)